src/.nuget/Microsoft.NET.Sdk.IL/build/Microsoft.NET.Sdk.IL.targets

```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+***********************************************************************************************
+Microsoft.NET.Sdk.IL.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved. 
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildThisFileFullPath);$(MSBuildAllProjects)</MSBuildAllProjects>
+    <DefaultLanguageSourceExtension>.il</DefaultLanguageSourceExtension>
+    <Language>IL</Language>
+    <TargetRuntime>Managed</TargetRuntime>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('windows'))">win</_OSPlatform>
+    <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('linux'))">linux</_OSPlatform>
+    <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('osx'))">osx</_OSPlatform>
+    <_OSArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture)</_OSArchitecture>
+
+    <MicrosoftNetCoreIlasmPackageRuntimeId Condition="'$(MicrosoftNetCoreIlasmPackageRuntimeId)' == ''">$(_OSPlatform)-$(_OSArchitecture.ToLower())</MicrosoftNetCoreIlasmPackageRuntimeId>
```

Comment:
Do we have an existing API to get `linux-musl` as the OSPlatform?

Reply:
Looking at the current runtime.json for `ilasm`, the same question applies to `rhel.6` and `tizen.4.0.0`

Reply:
I believe `IsOSPlatform('linux') == true` is still correct here.  We are still on `linux`, we just happen to be using a different `libc`. (And thus, requiring different native assets)

Reply:
It appears in corefx, the RID is always passed in from the outside of the build:

https://github.com/dotnet/corefx/search?q=linux-musl&unscoped_q=linux-musl

The only place it is actually detected is:

https://github.com/dotnet/corefx/blob/a53adfc55bcb028d36a8647e45c66f7be5374f92/init-tools.sh#L97

Reply:
The targets are configured such that `CoreFX` can explicitly pass in the RID when consuming the .NET Sdk....

However, it will not work automatically for `linux-musl`, `rhel.6`, or `tizen.4.0.0` without some mechanism to get that portion of the RID.

Reply:
@weshaggard, do you think this is sufficient for the current use-case in CoreFX/CoreCLR?

We can track fixing it more generally when/if the proposal to support detecting the current RID at runtime comes through.

Reply:
> we won't even get past CI because the linux-musl leg will fail.

For this, (on alpine, tizen, rhel6) I was indicating that we would continue passing it in from outside the build, as we are today.

An alternative would be to have the .targets file duplicate this logic: https://github.com/dotnet/corefx/blob/a53adfc55bcb028d36a8647e45c66f7be5374f92/init-tools.sh#L94

Reply:
Ah I missed that connection. In that case I think we can live with it for now so long as we can actually consume it I think we should be good.

Reply:
I've tested locally and everything is working correctly when explicitly passing in the RID.

Reply:
Some tools use `ldd --version` to figure out the libc flavor. Based on that, there could be an MSBuild target to detect glibc vs. musl-libc, for example:

Inspired by https://github.com/justjake/encabulator/blob/34b587b/unison/linux_env.go

```xml
<?xml version="1.0" encoding="utf-8"?>
<Project>
  <Target Name="DetectLibcFlavor" Condition="!$([MSBuild]::IsOSPlatform('windows'))">
    <Exec Command="ldd --version" ConsoleToMSBuild="true" ContinueOnError="true" StandardOutputImportance="low" StandardErrorImportance="low">
      <Output TaskParameter="ConsoleOutput" PropertyName="LddOutput" />
    </Exec>

    <Message Importance="high" Text="using glibc"
      Condition=" $([System.Text.RegularExpressions.Regex]::IsMatch($(LddOutput), `glibc|GNU libc|EGLIBC`)) " />
    <Message Importance="high" Text="using musl-libc"
      Condition=" $([System.Text.RegularExpressions.Regex]::IsMatch(`$(LddOutput)`, `musl`)) " />
  </Target>
</Project>
```

Tested with `dotnet msbuild DetectLibcFlavor.proj`.

Sample `ldd --version` output on Alpine:

```
musl libc (x86_64)
Version 1.1.18
Dynamic Program Loader
Usage: ldd [options] [--] pathname
```

and on Ubuntu:

```
ldd (Ubuntu EGLIBC 2.15-0ubuntu10.7) 2.15
Copyright (C) 2012 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

(would be super helpful if MSBuild provides information about libc via some intrincis method next to `[MSBuild]::IsOSPlatform` and friends, which we have today)